Goal: Export USER and SHELL as well as HOME.
Fixes: maybe #11003 and #11189

Status wrt upstream: It should be proposed to upstream.

Index: shadow-4.0.13/src/su.c
===================================================================
--- shadow-4.0.13.orig/src/su.c	2005-10-04 08:06:34.164595165 +0200
+++ shadow-4.0.13/src/su.c	2005-10-04 08:07:14.573361562 +0200
@@ -668,8 +668,11 @@
 	if (fakelogin)
 		setup_env (&pwent);
 #if 1				/* Suggested by Joey Hess. XXX - is this right?  */
-	else
+	else {
 		addenv ("HOME", pwent.pw_dir);
+		addenv ("USER", pwent.pw_name);
+		addenv ("SHELL", pwent.pw_shell);
+	}
 #endif
 
 	/*
